### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by SyamBabu-M**

### CODE REVIEW

File: .github/workflows/build-production-deploy-s3.yml [hunk 1]
```diff
@@ -17,4 +17,14 @@ jobs:
...
       - run: npm install
       - run: npm run build:production
-      - run: aws s3 sync ./build s3://${{env.WEBHOSTING_BUCKET_NAME}}/ --acl public-read --delete --cache-control max-age=3600
+      - run: |
+          current_version=$(node -p "require('./package.json').version")
+          major_version=$(echo $current_version | cut -d. -f1)
+
+          aws s3 sync ./build s3://${{env.WEBHOSTING_BUCKET_NAME}}/v${major_version}/ --acl public-read --cache-control max-age=3600
+
+          previous_major_version=$((major_version - 2))
```
Comment (saumier) [inline .github/workflows/build-production-deploy-s3.yml:26]: @SyamBabu-M Please leave all major versions. Do not delete major_version - 2.

File: .github/workflows/build-production-deploy-s3.yml [hunk 2]
```diff
@@ -17,4 +17,14 @@ jobs:
       - uses: actions/checkout@v2
       - run: npm install
       - run: npm run build:production
-      - run: aws s3 sync ./build s3://${{env.WEBHOSTING_BUCKET_NAME}}/ --acl public-read --delete --cache-control max-age=3600
+      - run: |
+          current_version=$(node -p "require('./package.json').version")
+          major_version=$(echo $current_version | cut -d. -f1)
+
+          aws s3 sync ./build s3://${{env.WEBHOSTING_BUCKET_NAME}}/v${major_version}/ --acl public-read --cache-control max-age=3600
```
Comment (saumier) [inline .github/workflows/build-production-deploy-s3.yml:24]: I think you are missing a delete command to remove old files from the directory. Try  `--acl public-read --cache-control max-age=3600 --delete `

**Review by saumier — CHANGES_REQUESTED** [review]

File: config-overrides.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable no-undef */
+const { version } = require('./package.json');
+module.exports = function override(config, env) {
```
Comment (saumier) [inline config-overrides.js:5]: @SyamBabu-M I wondering why you are changing the name of the .js and .css files as well as the path.  I think changing both the filenames and the path is redundant. I propose you change the path (by adding `v0`) but keep the filenames for .js and .css without any version.  Let me know if you a reason to add the version to the filenames.

**Review by saumier — CHANGES_REQUESTED** [review]

File: config-overrides.js (hunk 3 above)
Comment (SyamBabu-M) [reply config-overrides.js:5]: I agree. It seems redundant to add the version number to both the directory and .js and CSS files. I will update the .js and .css files to widget.js and widget.css respectively

**Review by saumier — APPROVED** [review]

**Pull request merged**